M2/Macaulay2/packages/Permutations.m2

```diff
@@ -0,0 +1,383 @@
+newPackage(
+    "Permutations",
+    AuxiliaryFiles => true,
+    Version => "1.0", 
+    Date => "August 7, 2024",
+    Keywords => {"Combinatorics"},
+    Authors => {
+        {Name => "Sean Grate", 
+         Email => "sean.grate@auburn.edu", 
+         HomePage => "https://seangrate.com/"}
+    },
+    Headline => "functions for working with permutations"
+)
+
+export {
+    -- types
+    "Permutation",
+    -- methods
+    "permutation",
+    "toMatrix",
+    "cycleDecomposition",
+    "cycleType",
+    "ascents",
+    "descents",
+    "ascendingRuns",
+    "descendingRuns",
+    "exceedances",
+    "saliances",
+    "records",
+    "avoidsPattern",
+    "avoidsPatterns",
+    "isVexillary",
+    "isCartwrightSturmfels",
+    "isCDG",
+    "foataBijection",
+    "ord",
+    "sign",
+    "isEven",
+    "isOdd",
+    "isDerangement",
+    "fixedPoints",
+    "inversions",
+    -- symbols
+    "Weak"
+}
+
+-----------------------------------------------------------------------------
+-- **CODE** --
+-----------------------------------------------------------------------------
+------------------------------------
+-- Local utilities
+------------------------------------
+to1Index := w -> (w / (i -> i+1))
+to0Index := w -> (w / (i -> i-1))
+
+------------------------------------
+-- Permutation type declarations and basic constructors
+------------------------------------
+Permutation = new Type of VisibleList
+Permutation.synonym = "permutation"
+
+new Permutation from VisibleList := (typeofPermutation,w) -> w
+
+permutation = method()
+permutation List := Permutation => w -> new Permutation from w
```

Comment:
If you want, I think it would be reasonable to move `permutations`, `inversePermutation`, and (possibly?) `uniquePermutations` into this package (along with their tests and documentation), and comment that they were moved from Core. This may break some tests in other packages, in which case we will add this package as their dependency.

Reply:
I think this make sense and I'll look into it.